spring-integration-jmx/src/main/java/org/springframework/integration/monitor/IntegrationMBeanExporter.java

```diff
@@ -294,6 +299,14 @@ public void setEmbeddedValueResolver(StringValueResolver resolver) {
 		this.embeddedValueResolver = resolver;
 	}
 
+	/**
+	 * Set a metrics factory.
+	 * @param metricsFactory the factory.
```

Comment:
Will remove `applyAdvice()`.

To be honest, I don't remember the TODO; I will have to do some research,